src/routes/embedded/auth/auth-email-otp/auth-email-otp.view.tsx

```diff
@@ -0,0 +1,196 @@
+
+export function AuthEmailOtpEmbeddedView() {
+  const { navigate } = useLocation();
+  const { email } = useSearchParams<{ email: string }>();
+  const { authStatus, authenticate } = useEmbedded();
+
+  const [_, setPreferredEmailAuth] = useStorage<PreferredEmailAuth | undefined>({
+    key: StorageKeys.CONNECT.AUTH.PREFERRED_EMAIL_AUTH,
+    instance: PersistentStorage,
+  });
+
+  // Loading state:
+
+  const [isResending, setIsResending] = useState(false);
+  const [isVerifying, setIsVerifying] = useState(false);
+  const isViewLoading =
+    authStatus === "unknown" || authStatus === "loading" || authStatus === "authLoading" || isVerifying;
+  const areButtonsDisabled = isViewLoading || isResending;
+
+  // Code input:
+
+  const codeInputRef = useRef<CodeInputHandle>();
+  const [isComplete, setIsComplete] = useState(false);
+
+  const handleCodeChange = useCallback((_, isComplete: boolean) => {
+    setIsComplete(isComplete);
+  }, []);
+
+  // Code retrieval:
+
+  const { fn: signInWithOtp, cooldownSeconds } = useCooldownCallback(
+    async (showConfirmationToast: boolean) => {
+      try {
+        if (codeInputRef.current) codeInputRef.current.clear();
+
+        setIsResending(true);
+
+        const supabase = await getSupabaseClient();
+
+        const { error } = await supabase.auth.signInWithOtp({ email });
+
+        if (error) {
+          toast.error(getFriendlyAuthErrorMessage(error, "Error sending email access code"));
+          return;
+        }
+
+        setOtpAvailable();
+
+        if (showConfirmationToast) toast.success("Email access code resent successfully");
+
+        // Note we don't need to check if the email is verified or not. Once the user signs in using the OTP code, their
+        // email is verified if it wasn't already and the router redirects them to the right page (add wallet, backup reminder...).
+      } catch (error) {
+        toast.error(getFriendlyAuthErrorMessage(error, "Error sending email access code"));
+      } finally {
+        setIsResending(false);
+      }
+    },
+    {
+      key: StorageKeys.CONNECT.AUTH.LAST_OTP_EMAIL,
+      cooldownDuration: OPT_COOLDOWN_DURATION_SEC,
+    },
+  );
+
+  useAsyncEffect(async () => {
+    try {
+      if (await checkNeedsNewOtp()) signInWithOtp(false);
+    } catch {
+      // In case `LAST_OTP_EMAIL` is already be set in `localStorage` and the required time hasn't passed yet (so this call will throw an error).
+    }
+  }, [signInWithOtp]);
+
+  // Handlers:
+
+  const handleVerifyOtp = useCallback(
+    async (e: React.FormEvent) => {
+      e.preventDefault();
+
+      if (!email || isVerifying) return;
+
+      const otpCode = codeInputRef.current.getCode();
+
+      if (otpCode.length !== OTP_LENGTH) {
+        toast.error(`Please enter all ${OTP_LENGTH} digits of the access code`);
+        return;
+      }
+
+      setIsVerifying(true);
+
+      try {
+        await authenticate({
```

Comment:
This now takes care of both sign ups and OTP sign ins, so the sign up view is gone.